pkg/process/info.go

```diff
@@ -189,7 +217,168 @@ type Info struct {
 	//   * "/proc/%d/root/tmp/perf-%d.map" or "/proc/%d/root/tmp/perf-%d.dump" for PerfMaps
 	//   * "/proc/%d/root/jit-%d.dump" for JITDUMP
 	// - Unwind Information
-	Mappings Mappings
+	Interpreter *Interpreter
+	Mappings    Mappings
+}
+
+// fetchRubyInterpreterInfo receives a process pid and memory mappings and
+// figures out whether it might be a Ruby interpreter. In that case, it
+// returns an `Interpreter` structure with the data that is needed by rbperf
+// (https://github.com/javierhonduco/rbperf) to walk Ruby stacks.
+func fetchRubyInterpreterInfo(pid int, mappings Mappings) (*Interpreter, error) {
```

Comment:
♻️ I think this deserves its own package, and we have a `runtime` package for this. Let's consider moving it.

Reply:
If it's ok with you, I would like to do this in a forthcoming PR

Reply:
As long as we don't forget about it 👍

Reply:
💯